tests/code_update.py

```diff
@@ -51,6 +51,180 @@ def test_verify_quotes(network, args):
 
     return network
 
+@reqs.description("Node with correct security policy joins successfully")
+def test_add_node_with_correct_security_policy(network, args):
+
+    if not IS_SNP:
+        LOG.warning("Skipping test_add_node_with_correct_security_policy with non SNP platform")
+        return network
+
+    replacement_package = (
+        "samples/apps/logging/liblogging"
+        if args.package == "libjs_generic"
+        else "libjs_generic"
+    )
+
+    primary, others = network.find_nodes()
+    prev_nodes = [primary, *others]
+    new_node = network.create_node("local://localhost")
+    network.join_node(new_node, replacement_package, args, timeout=3)
+    network.trust_node(new_node, args)
+    primary, others = network.find_nodes()
+    nodes = [primary, *others]
+    assert len(nodes) == len(prev_nodes) + 1, "New node not added to network"
```

Comment:
If `network.trust_node(new_node, args)` succeeds (i.e. doesn't throw), then you know that the new has successfully joined the network so need to check for this here.

Reply:
Okay removed some of the checks